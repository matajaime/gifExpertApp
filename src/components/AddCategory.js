import React, { useState } from "react";
import PropTypes from 'prop-types';

export const AddCategory = ({addCategory}) => {
    
    const [inputValue, setInputValue] = useState('');
    
    const handleInputChange = (event) => {
        setInputValue(event.target.value);
    };

    const handleSubmit = (event) => {
        event.preventDefault();
        //console.log(event);
        if ( inputValue != '' ) {
            addCategory(inputValue);
            setInputValue('');    
        }
    };

    const cleanInput = (event) => {
        event.target.value = '';
    }

    return (
        <form onSubmit={handleSubmit}>
        <h4>Add Category</h4>
        <input type="text" value={inputValue} placeholder="Ingresa una categoria" 
        onChange={handleInputChange} onFocus={cleanInput}/>
        </form>
    );
};

AddCategory.propTypes = {
    addCategory: PropTypes.func.isRequired
}
