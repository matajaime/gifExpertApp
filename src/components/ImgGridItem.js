import React from "react";
import PropTypes from 'prop-types';

export const ImgGridItem = ({img}) => {
    //console.log(img);
    return (
        <div className="card animate__animated animate__fadeInDownBig">
            <img src={img.url} alt={img.title} />
            <p>{img.title}</p>
        </div>
    );
};

ImgGridItem.propTypes = {
    img : PropTypes.object.isRequired
}

