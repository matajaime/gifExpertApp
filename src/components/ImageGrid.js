import React, {useEffect, useState} from "react";
import { ImgGridItem } from "./ImgGridItem";
import { useFetchGifs } from "../hooks/useFetchGifs";
import PropTypes from 'prop-types';

export const ImageGrid = ({category}) => {
    
    //const[gifs, setGifs] = useState([]);
    //useEffect( () => {
    //    getGifs(category).then( setGifs );
    //}, [ category ]);

    const {loading, data:images} = useFetchGifs(category);

    return (
        <>
        <h3>{category}</h3>
        { loading ? <p>Cargando...</p> : 
        <div className="card-grid">
            { images.map((gif)=>{
                return <ImgGridItem key={gif.id} img={gif} />;
            })}
        </div>}
        </>
    )
};

ImageGrid.propTypes = {
    category: PropTypes.string.isRequired
}
