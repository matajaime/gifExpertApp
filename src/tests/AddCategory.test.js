import React from 'react';
import { shallow } from 'enzyme';
import { AddCategory } from '../components/AddCategory';
import '@testing-library/jest-dom';

describe('Tests AddCategory component',()=>{
    test('Snapshot test initial state',()=>{
        const funcAddCategory = jest.fn();
        const wrapper = shallow(<AddCategory addCategory={funcAddCategory}/>);
        expect(wrapper).toMatchSnapshot();
    });
    test('Debe de cambiar el input',()=>{
        const funcAddCategory = jest.fn();
        const wrapper = shallow(<AddCategory addCategory={funcAddCategory}/>);
        let input = wrapper.find('input');
        input.simulate('change', {
            target: {
                value: 'MiCategoria'
            }
        });
        input = wrapper.find('input');
        expect(input.props().value).toBe('MiCategoria');
    });
    test('Verficiar submit sin datos - no debe de llamar addCategory',()=>{
        const funcAddCategory = jest.fn();
        const wrapper = shallow(<AddCategory addCategory={funcAddCategory}/>);
        const forma = wrapper.find('form');
        forma.simulate('submit', {
            preventDefault: ()=>{}
        })
        expect( funcAddCategory ).not.toHaveBeenCalled();
    });
    test('Debe llamar addCategory y limpiar la caja de texto',()=>{
        const funcAddCategory = jest.fn();
        const wrapper = shallow(<AddCategory addCategory={funcAddCategory}/>);
        let input = wrapper.find('input');
        input.simulate('change', {
            target: {
                value: 'MiCategoria'
            }
        });
        const forma = wrapper.find('form');
        forma.simulate('submit', {
            preventDefault: ()=>{}
        })
        expect( funcAddCategory ).toHaveBeenCalled();
        //expect( funcAddCategory ).toHaveBeenCalledWith( expect.any(Function) );
        expect( funcAddCategory ).toHaveBeenCalledWith( expect.any(String) );
        input = wrapper.find('input');
        expect(input.props().value).toBe('');
    });
});