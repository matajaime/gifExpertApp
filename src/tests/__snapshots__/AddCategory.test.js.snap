// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests AddCategory component Snapshot test initial state 1`] = `
<form
  onSubmit={[Function]}
>
  <h4>
    Add Category
  </h4>
  <input
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Ingresa una categoria"
    type="text"
    value=""
  />
</form>
`;
